corehq/apps/app_manager/translations.py

```diff
@@ -480,34 +497,43 @@ def update_case_list_translations(sheet, rows, app):
             ))
             continue
 
-        # The logic for updating a mapping and updating a MappingItem and a
-        # DetailColumn is almost the same. So, we smush the two together.
-        for index, translation_row in enumerate([row] + row.get("mappings", [])):
-            ok_to_delete_translations = has_at_least_one_translation(
-                translation_row, 'default', app.langs)
+        def _update_translation(row, language_dict, require_translation=True):
+            ok_to_delete_translations = (
+                not require_translation or has_at_least_one_translation(
+                    row, 'default', app.langs
+                ))
             if ok_to_delete_translations:
                 for lang in app.langs:
-                    translation = translation_row['default_%s' % lang]
-                    if index == 0:
-                        # For DetailColumns
-                        language_dict = detail.header
-                    else:
-                        # For MappingItems
-                        language_dict = detail['enum'][index - 1].value
-
-                    if translation:
-                        language_dict[lang] = translation
-                    else:
-                        if lang in language_dict:
-                            del language_dict[lang]
+                    translation = row['default_%s' % lang]
+                if translation:
```

Comment:
Oh good catch. The `if/else` is supposed to be indented. This causes it to only update/delete the last language's translations of the case list. Added a new test that demonstrates the bug and a commit to fix it.